models/external-accounts.js

```diff
@@ -11,4 +11,33 @@ const addExternalAccountData = async (data) => {
   }
 };
 
-module.exports = { addExternalAccountData };
+const fetchExternalAccountData = async (query, param) => {
+  try {
+    const userExternalAccountData = [];
+
+    let externalAccountData;
+    externalAccountData = await externalAccountsModel.where("token", "==", param);
+
+    Object.keys(query).forEach((key) => {
+      if (key === "type") {
+        externalAccountData = externalAccountData.where(key, "==", query?.key);
+      }
+    });
```

Comment:
The `forEach` only runs for `key`  `type`, why do we need a loop here?

Reply:
It was there because we can have more queries in future and then the if check can be removed. But yeah we can take it of it in future and I have removed the loop for now